### DISCUSSION TITLE
hi, how to query all ecclasses in an imodel?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#957] [Q&A] [answered]
hi, how to query all ecclasses in an imodel? thanks.
I want to get some user-defined classes. I need to get all ecclasses first, and then filter them according to some rules.

**Comment by grigasp** [comment, marked as answer]
You can use the `ECDbMeta` ECSchema for that:
```
SELECT * FROM meta.ECClassDef
```

**Reply by shenshengyi** [reply]
Thank you very much, your answer helped me a lot
